src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -64,28 +60,60 @@ public ScriptScope Scope
 
         /// <summary>
         /// Maps a regex to a particular python type.  Useful for matching things like dicts,
-        /// floats, strings, etc.  Initialized by 
+        /// floats, strings, etc.  Initialized by
         /// </summary>
         public Dictionary<string, Type> RegexToType = new Dictionary<string, Type>();
+        
+        /// <summary>
+        /// Maps a basic variable regex to a basic python type.
+        /// </summary>
+        public List<Tuple<Regex, Type>> BasicVariableTypes;
+        
+        /// <summary>
+        /// Tracks already referenced CLR modules
+        /// </summary>
+        public HashSet<string> clrModules { get; set; }
+        
+        /// <summary>
+        /// Keeps track of failed statements to avoid poluting the log
+        /// </summary>
+        public Dictionary<string, int> badStatements { get; set; }
 
         /// <summary>
         /// A bunch of regexes for use in introspaction
         /// </summary>
-        public static string commaDelimitedVariableNamesRegex = @"(([0-9a-zA-Z_]+,?\s*)+)";
-        public static string variableName = @"([0-9a-zA-Z_]+(\.[a-zA-Z_0-9]+)*)";   
-        public static string doubleQuoteStringRegex = "(\"[^\"]*\")";
-        public static string singleQuoteStringRegex = "(\'[^\']*\')";
+        public static string commaDelimitedVariableNamesRegex = @"(([0-9a-zA-Z_]+,?\s?)+)";
+        public static string variableName = @"([0-9a-zA-Z_]+(\.[a-zA-Z_0-9]+)*)";
+        public static string quotesStringRegex = "[\"']([^\"']*)[\"']";
         public static string arrayRegex = "(\\[.*\\])";
         public static string spacesOrNone = @"(\s*)";
         public static string atLeastOneSpaceRegex = @"(\s+)";
-        public static string equals = @"(=)";
+        public static string equalsRegex = @"(=)";
         public static string dictRegex = "({.*})";
         public static string doubleRegex = @"([-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?)";
         public static string intRegex = @"([-+]?\d+)[\s\n]*$";
         public static string basicImportRegex = @"(import)";
         public static string fromImportRegex = @"^(from)";
-
-#endregion
+        
+        private static readonly Regex MATCH_LAST_NAMESPACE = new Regex(@"[\w.]+$", RegexOptions.Compiled);
```

Comment:
I don't have a problem with the readonly fields being uppercase - but curious - is this from a specific style guide?